decompose-router/src/commonMain/kotlin/io/github/xxfast/decompose/router/Router.kt

```diff
@@ -51,16 +53,31 @@ fun <C : Parcelable> rememberRouter(
   stack: List<C>,
   handleBackButton: Boolean = true
 ): Router<C> {
-  val navigator: StackNavigation<C> = remember { StackNavigation() }
-  val childStackState: State<ChildStack<C, ComponentContext>> = rememberChildStack(
-    source = navigator,
-    initialStack = { stack },
-    key = key.toString(), // Has to use strings for Android 😢
-    handleBackButton = handleBackButton,
-    type = type,
-  )
+  val routerContext = LocalRouterContext.current
+  val keyStr = key.toString()
 
-  return remember { Router(navigator = navigator, stack = childStackState) }
+  return remember {
```

Comment:
@arkivanov Do you think we need to key this remember with our  `key`
```kotlin
return remember(key) {
    routerContext.getOrCreate(key = keyStr) { .. }
}
```

Reply:
Yeah, you can use keyStr, the same as for getOrCreate